### DISCUSSION TITLE
class contains c array, compilation error is reported

### DISCUSSION THREAD

**Discussion by huqin-RM** [pybind/pybind11#4948] [Q&A]
class Ctest
{
public:
    double m_vdMat[3][3];
};

PYBIND11_EMBEDDED_MODULE(fast_calc,m){
    py::class_<Ctest>(m, "Ctest")
            .def(py::init<>())
            .def_readwrite("m_vdMat", &CIntrinsicsM::m_vdMat);

}

/usr/local/include/pybind11/pybind11.h:1717:56: error: invalid array assignment
 1717 |             fset([pm](type &c, const D &value) { c.*pm = value; }, is_method(*this));
      |                                                  ~~~~~~^~~~~~~
/usr/local/include/pybind11/pybind11.h: In instantiation of ‘void pybind11::cpp_function::initialize(Func&&, Return (*)(Args ...), const Extra& ...) [with Func = pybind11::class_<type_, options>::def_readwrite(const char*, D C::*, const Extra& ...) [with C = Ctest; D = double [3][3]; Extra = {}; type_ = Ctest; options = {}]::<lambda(const type&)>; Return = const double (&)[3][3]; Args = {const Ctest&}; Extra = {pybind11::is_method}]’:
/usr/local/include/pybind11/pybind11.h:145:9:   required from ‘pybind11::cpp_function::cpp_function(Func&&, const Extra& ...) [with Func = pybind11::class_<type_, options>::def_readwrite(const char*, D C::*, const Extra& ...) [with C = Ctest; D = double [3][3]; Extra = {}; type_ = Ctest; options = {}]::<lambda(const type&)>; Extra = {pybind11::is_method}; <template-parameter-1-3> = void]’
/usr/local/include/pybind11/pybind11.h:1716:22:   required from ‘pybind11::class_<type_, options>& pybind11::class_<type_, options>::def_readwrite(const char*, D C::*, const Extra& ...) [with C = Ctest; D = double [3][3]; Extra = {}; type_ = Ctest; options = {}]’
1.cpp:150:54:   required from here
/usr/local/include/pybind11/pybind11.h:296:40: error: no matching function for call to ‘pybind11::detail::type_caster<double, void>::cast(const double [3][3], pybind11::return_value_policy&, pybind11::handle&)’
  296 |                 result = cast_out::cast(
      |                          ~~~~~~~~~~~~~~^
  297 |                     std::move(args_converter).template call<Return, Guard>(cap->f),
      |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  298 |                     policy,
      |                     ~~~~~~~             
  299 |                     call.parent);
      |                     ~~~~~~~~~~~~        
In file included from /usr/local/include/pybind11/attr.h:14,
                 from /usr/local/include/pybind11/detail/class.h:12,
                 from /usr/local/include/pybind11/pybind11.h:13,
                 from /usr/local/include/pybind11/embed.h:12,
                 from 1.cpp:1:
/usr/local/include/pybind11/cast.h:208:5: note: candidate: ‘template<class U> static typename std::enable_if<std::is_floating_point<U>::value, pybind11::handle>::type pybind11::detail::type_caster<T, typename std::enable_if<(std::is_arithmetic<_Tp>::value && (! pybind11::detail::negation<std::is_same<pybind11::detail::bools<pybind11::detail::negation<std::is_same<CharT, char> >::value, pybind11::detail::negation<std::is_same<CharT, char16_t> >::value, pybind11::detail::negation<std::is_same<CharT, char32_t> >::value, pybind11::detail::negation<std::is_same<CharT, wchar_t> >::value, true>, pybind11::detail::bools<true, pybind11::detail::negation<std::is_same<CharT, char> >::value, pybind11::detail::negation<std::is_same<CharT, char16_t> >::value, pybind11::detail::negation<std::is_same<CharT, char32_t> >::value, pybind11::detail::negation<std::is_same<CharT, wchar_t> >::value> > >::value)), void>::type>::cast(U, pybind11::return_value_policy, pybind11::handle) [with U = U; T = double]’
  208 |     cast(U src, return_value_policy /* policy */, handle /* parent */) {
      |     ^~~~
/usr/local/include/pybind11/cast.h:208:5: note:   template argument deduction/substitution failed:
/usr/local/include/pybind11/cast.h: In substitution of ‘template<class U> static typename std::enable_if<std::is_floating_point<_Tp>::value, pybind11::handle>::type pybind11::detail::type_caster<double, void>::cast<U>(U, pybind11::return_value_policy, pybind11::handle) [with U = const double (*)[3]]’:
/usr/local/include/pybind11/pybind11.h:296:40:   required from ‘void pybind11::cpp_function::initialize(Func&&, Return (*)(Args ...), const Extra& ...) [with Func = pybind11::class_<type_, options>::def_readwrite(const char*, D C::*, const Extra& ...) [with C = Ctest; D = double [3][3]; Extra = {}; type_ = Ctest; options = {}]::<lambda(const type&)>; Return = const double (&)[3][3]; Args = {const Ctest&}; Extra = {pybind11::is_method}]’
/usr/local/include/pybind11/pybind11.h:145:9:   required from ‘pybind11::cpp_function::cpp_function(Func&&, const Extra& ...) [with Func = pybind11::class_<type_, options>::def_readwrite(const char*, D C::*, const Extra& ...) [with C = Ctest; D = double [3][3]; Extra = {}; type_ = Ctest; options = {}]::<lambda(const type&)>; Extra = {pybind11::is_method}; <template-parameter-1-3> = void]’
/usr/local/include/pybind11/pybind11.h:1716:22:   required from ‘pybind11::class_<type_, options>& pybind11::class_<type_, options>::def_readwrite(const char*, D C::*, const Extra& ...) [with C = Ctest; D = double [3][3]; Extra = {}; type_ = Ctest; options = {}]’
1.cpp:150:54:   required from here
/usr/local/include/pybind11/cast.h:208:5: error: no type named ‘type’ in ‘struct std::enable_if<false, pybind11::handle>’
/usr/local/include/pybind11/pybind11.h: In instantiation of ‘void pybind11::cpp_function::initialize(Func&&, Return (*)(Args ...), const Extra& ...) [with Func = pybind11::class_<type_, options>::def_readwrite(const char*, D C::*, const Extra& ...) [with C = Ctest; D = double [3][3]; Extra = {}; type_ = Ctest; options = {}]::<lambda(const type&)>; Return = const double (&)[3][3]; Args = {const Ctest&}; Extra = {pybind11::is_method}]’:
/usr/local/include/pybind11/pybind11.h:145:9:   required from ‘pybind11::cpp_function::cpp_function(Func&&, const Extra& ...) [with Func = pybind11::class_<type_, options>::def_readwrite(const char*, D C::*, const Extra& ...) [with C = Ctest; D = double [3][3]; Extra = {}; type_ = Ctest; options = {}]::<lambda(const type&)>; Extra = {pybind11::is_method}; <template-parameter-1-3> = void]’
/usr/local/include/pybind11/pybind11.h:1716:22:   required from ‘pybind11::class_<type_, options>& pybind11::class_<type_, options>::def_readwrite(const char*, D C::*, const Extra& ...) [with C = Ctest; D = double [3][3]; Extra = {}; type_ = Ctest; options = {}]’
1.cpp:150:54:   required from here
/usr/local/include/pybind11/cast.h:216:5: note: candidate: ‘template<class U> static typename std::enable_if<(((! std::is_floating_point<U>::value) && std::is_signed<U>::value) && (sizeof (U) <= sizeof (long int))), pybind11::handle>::type pybind11::detail::type_caster<T, typename std::enable_if<(std::is_arithmetic<_Tp>::value && (! pybind11::detail::negation<std::is_same<pybind11::detail::bools<pybind11::detail::negation<std::is_same<CharT, char> >::value, pybind11::detail::negation<std::is_same<CharT, char16_t> >::value, pybind11::detail::negation<std::is_same<CharT, char32_t> >::value, pybind11::detail::negation<std::is_same<CharT, wchar_t> >::value, true>, pybind11::detail::bools<true, pybind11::detail::negation<std::is_same<CharT, char> >::value, pybind11::detail::negation<std::is_same<CharT, char16_t> >::value, pybind11::detail::negation<std::is_same<CharT, char32_t> >::value, pybind11::detail::negation<std::is_same<CharT, wchar_t> >::value> > >::value)), void>::type>::cast(U, pybind11::return_value_policy, pybind11::handle) [with U = U; T = double]’
  216 |     cast(U src, return_value_policy /* policy */, handle /* parent */) {
      |     ^~~~
/usr/local/include/pybind11/cast.h:216:5: note:   template argument deduction/substitution failed:
/usr/local/include/pybind11/cast.h: In substitution of ‘template<class U> static typename std::enable_if<(((! std::is_floating_point<_Tp>::value) && std::is_signed<_Tp>::value) && (sizeof (U) <= sizeof (long int))), pybind11::handle>::type pybind11::detail::type_caster<double, void>::cast<U>(U, pybind11::return_value_policy, pybind11::handle) [with U = const double (*)[3]]’:
/usr/local/include/pybind11/pybind11.h:296:40:   required from ‘void pybind11::cpp_function::initialize(Func&&, Return (*)(Args ...), const Extra& ...) [with Func = pybind11::class_<type_, options>::def_readwrite(const char*, D C::*, const Extra& ...) [with C = Ctest; D = double [3][3]; Extra = {}; type_ = Ctest; options = {}]::<lambda(const type&)>; Return = const double (&)[3][3]; Args = {const Ctest&}; Extra = {pybind11::is_method}]’
/usr/local/include/pybind11/pybind11.h:145:9:   required from ‘pybind11::cpp_function::cpp_function(Func&&, const Extra& ...) [with Func = pybind11::class_<type_, options>::def_readwrite(const char*, D C::*, const Extra& ...) [with C = Ctest; D = double [3][3]; Extra = {}; type_ = Ctest; options = {}]::<lambda(const type&)>; Extra = {pybind11::is_method}; <template-parameter-1-3> = void]’
/usr/local/include/pybind11/pybind11.h:1716:22:   required from ‘pybind11::class_<type_, options>& pybind11::class_<type_, options>::def_readwrite(const char*, D C::*, const Extra& ...) [with C = Ctest; D = double [3][3]; Extra = {}; type_ = Ctest; options = {}]’
1.cpp:150:54:   required from here
/usr/local/include/pybind11/cast.h:216:5: error: no type named ‘type’ in ‘struct std::enable_if<false, pybind11::handle>’
/usr/local/include/pybind11/pybind11.h: In instantiation of ‘void pybind11::cpp_function::initialize(Func&&, Return (*)(Args ...), const Extra& ...) [with Func = pybind11::class_<type_, options>::def_readwrite(const char*, D C::*, const Extra& ...) [with C = Ctest; D = double [3][3]; Extra = {}; type_ = Ctest; options = {}]::<lambda(const type&)>; Return = const double (&)[3][3]; Args = {const Ctest&}; Extra = {pybind11::is_method}]’:
/usr/local/include/pybind11/pybind11.h:145:9:   required from ‘pybind11::cpp_function::cpp_function(Func&&, const Extra& ...) [with Func = pybind11::class_<type_, options>::def_readwrite(const char*, D C::*, const Extra& ...) [with C = Ctest; D = double [3][3]; Extra = {}; type_ = Ctest; options = {}]::<lambda(const type&)>; Extra = {pybind11::is_method}; <template-parameter-1-3> = void]’
/usr/local/include/pybind11/pybind11.h:1716:22:   required from ‘pybind11::class_<type_, options>& pybind11::class_<type_, options>::def_readwrite(const char*, D C::*, const Extra& ...) [with C = Ctest; D = double [3][3]; Extra = {}; type_ = Ctest; options = {}]’
1.cpp:150:54:   required from here
/usr/local/include/pybind11/cast.h:224:5: note: candidate: ‘template<class U> static typename std::enable_if<(((! std::is_floating_point<U>::value) && std::is_unsigned<U>::value) && (sizeof (U) <= sizeof (long unsigned int))), pybind11::handle>::type pybind11::detail::type_caster<T, typename std::enable_if<(std::is_arithmetic<_Tp>::value && (! pybind11::detail::negation<std::is_same<pybind11::detail::bools<pybind11::detail::negation<std::is_same<CharT, char> >::value, pybind11::detail::negation<std::is_same<CharT, char16_t> >::value, pybind11::detail::negation<std::is_same<CharT, char32_t> >::value, pybind11::detail::negation<std::is_same<CharT, wchar_t> >::value, true>, pybind11::detail::bools<true, pybind11::detail::negation<std::is_same<CharT, char> >::value, pybind11::detail::negation<std::is_same<CharT, char16_t> >::value, pybind11::detail::negation<std::is_same<CharT, char32_t> >::value, pybind11::detail::negation<std::is_same<CharT, wchar_t> >::value> > >::value)), void>::type>::cast(U, pybind11::return_value_policy, pybind11::handle) [with U = U; T = double]’
  224 |     cast(U src, return_value_policy /* policy */, handle /* parent */) {
      |     ^~~~
/usr/local/include/pybind11/cast.h:224:5: note:   template argument deduction/substitution failed:
/usr/local/include/pybind11/cast.h: In substitution of ‘template<class U> static typename std::enable_if<(((! std::is_floating_point<_Tp>::value) && std::is_unsigned<T>::value) && (sizeof (U) <= sizeof (long unsigned int))), pybind11::handle>::type pybind11::detail::type_caster<double, void>::cast<U>(U, pybind11::return_value_policy, pybind11::handle) [with U = const double (*)[3]]’:
/usr/local/include/pybind11/pybind11.h:296:40:   required from ‘void pybind11::cpp_function::initialize(Func&&, Return (*)(Args ...), const Extra& ...) [with Func = pybind11::class_<type_, options>::def_readwrite(const char*, D C::*, const Extra& ...) [with C = Ctest; D = double [3][3]; Extra = {}; type_ = Ctest; options = {}]::<lambda(const type&)>; Return = const double (&)[3][3]; Args = {const Ctest&}; Extra = {pybind11::is_method}]’
/usr/local/include/pybind11/pybind11.h:145:9:   required from ‘pybind11::cpp_function::cpp_function(Func&&, const Extra& ...) [with Func = pybind11::class_<type_, options>::def_readwrite(const char*, D C::*, const Extra& ...) [with C = Ctest; D = double [3][3]; Extra = {}; type_ = Ctest; options = {}]::<lambda(const type&)>; Extra = {pybind11::is_method}; <template-parameter-1-3> = void]’
/usr/local/include/pybind11/pybind11.h:1716:22:   required from ‘pybind11::class_<type_, options>& pybind11::class_<type_, options>::def_readwrite(const char*, D C::*, const Extra& ...) [with C = Ctest; D = double [3][3]; Extra = {}; type_ = Ctest; options = {}]’
1.cpp:150:54:   required from here
/usr/local/include/pybind11/cast.h:224:5: error: no type named ‘type’ in ‘struct std::enable_if<false, pybind11::handle>’
/usr/local/include/pybind11/pybind11.h: In instantiation of ‘void pybind11::cpp_function::initialize(Func&&, Return (*)(Args ...), const Extra& ...) [with Func = pybind11::class_<type_, options>::def_readwrite(const char*, D C::*, const Extra& ...) [with C = Ctest; D = double [3][3]; Extra = {}; type_ = Ctest; options = {}]::<lambda(const type&)>; Return = const double (&)[3][3]; Args = {const Ctest&}; Extra = {pybind11::is_method}]’:
/usr/local/include/pybind11/pybind11.h:145:9:   required from ‘pybind11::cpp_function::cpp_function(Func&&, const Extra& ...) [with Func = pybind11::class_<type_, options>::def_readwrite(const char*, D C::*, const Extra& ...) [with C = Ctest; D = double [3][3]; Extra = {}; type_ = Ctest; options = {}]::<lambda(const type&)>; Extra = {pybind11::is_method}; <template-parameter-1-3> = void]’
/usr/local/include/pybind11/pybind11.h:1716:22:   required from ‘pybind11::class_<type_, options>& pybind11::class_<type_, options>::def_readwrite(const char*, D C::*, const Extra& ...) [with C = Ctest; D = double [3][3]; Extra = {}; type_ = Ctest; options = {}]’
1.cpp:150:54:   required from here
/usr/local/include/pybind11/cast.h:232:5: note: candidate: ‘template<class U> static typename std::enable_if<(((! std::is_floating_point<U>::value) && std::is_signed<U>::value) && (sizeof (U) > sizeof (long int))), pybind11::handle>::type pybind11::detail::type_caster<T, typename std::enable_if<(std::is_arithmetic<_Tp>::value && (! pybind11::detail::negation<std::is_same<pybind11::detail::bools<pybind11::detail::negation<std::is_same<CharT, char> >::value, pybind11::detail::negation<std::is_same<CharT, char16_t> >::value, pybind11::detail::negation<std::is_same<CharT, char32_t> >::value, pybind11::detail::negation<std::is_same<CharT, wchar_t> >::value, true>, pybind11::detail::bools<true, pybind11::detail::negation<std::is_same<CharT, char> >::value, pybind11::detail::negation<std::is_same<CharT, char16_t> >::value, pybind11::detail::negation<std::is_same<CharT, char32_t> >::value, pybind11::detail::negation<std::is_same<CharT, wchar_t> >::value> > >::value)), void>::type>::cast(U, pybind11::return_value_policy, pybind11::handle) [with U = U; T = double]’
  232 |     cast(U src, return_value_policy /* policy */, handle /* parent */) {
      |     ^~~~
/usr/local/include/pybind11/cast.h:232:5: note:   template argument deduction/substitution failed:
/usr/local/include/pybind11/cast.h: In substitution of ‘template<class U> static typename std::enable_if<(((! std::is_floating_point<_Tp>::value) && std::is_signed<_Tp>::value) && (sizeof (U) > sizeof (long int))), pybind11::handle>::type pybind11::detail::type_caster<double, void>::cast<U>(U, pybind11::return_value_policy, pybind11::handle) [with U = const double (*)[3]]’:
/usr/local/include/pybind11/pybind11.h:296:40:   required from ‘void pybind11::cpp_function::initialize(Func&&, Return (*)(Args ...), const Extra& ...) [with Func = pybind11::class_<type_, options>::def_readwrite(const char*, D C::*, const Extra& ...) [with C = Ctest; D = double [3][3]; Extra = {}; type_ = Ctest; options = {}]::<lambda(const type&)>; Return = const double (&)[3][3]; Args = {const Ctest&}; Extra = {pybind11::is_method}]’
/usr/local/include/pybind11/pybind11.h:145:9:   required from ‘pybind11::cpp_function::cpp_function(Func&&, const Extra& ...) [with Func = pybind11::class_<type_, options>::def_readwrite(const char*, D C::*, const Extra& ...) [with C = Ctest; D = double [3][3]; Extra = {}; type_ = Ctest; options = {}]::<lambda(const type&)>; Extra = {pybind11::is_method}; <template-parameter-1-3> = void]’
/usr/local/include/pybind11/pybind11.h:1716:22:   required from ‘pybind11::class_<type_, options>& pybind11::class_<type_, options>::def_readwrite(const char*, D C::*, const Extra& ...) [with C = Ctest; D = double [3][3]; Extra = {}; type_ = Ctest; options = {}]’
1.cpp:150:54:   required from here
/usr/local/include/pybind11/cast.h:232:5: error: no type named ‘type’ in ‘struct std::enable_if<false, pybind11::handle>’
/usr/local/include/pybind11/pybind11.h: In instantiation of ‘void pybind11::cpp_function::initialize(Func&&, Return (*)(Args ...), const Extra& ...) [with Func = pybind11::class_<type_, options>::def_readwrite(const char*, D C::*, const Extra& ...) [with C = Ctest; D = double [3][3]; Extra = {}; type_ = Ctest; options = {}]::<lambda(const type&)>; Return = const double (&)[3][3]; Args = {const Ctest&}; Extra = {pybind11::is_method}]’:
/usr/local/include/pybind11/pybind11.h:145:9:   required from ‘pybind11::cpp_function::cpp_function(Func&&, const Extra& ...) [with Func = pybind11::class_<type_, options>::def_readwrite(const char*, D C::*, const Extra& ...) [with C = Ctest; D = double [3][3]; Extra = {}; type_ = Ctest; options = {}]::<lambda(const type&)>; Extra = {pybind11::is_method}; <template-parameter-1-3> = void]’
/usr/local/include/pybind11/pybind11.h:1716:22:   required from ‘pybind11::class_<type_, options>& pybind11::class_<type_, options>::def_readwrite(const char*, D C::*, const Extra& ...) [with C = Ctest; D = double [3][3]; Extra = {}; type_ = Ctest; options = {}]’
1.cpp:150:54:   required from here
/usr/local/include/pybind11/cast.h:240:5: note: candidate: ‘template<class U> static typename std::enable_if<(((! std::is_floating_point<U>::value) && std::is_unsigned<U>::value) && (sizeof (U) > sizeof (long unsigned int))), pybind11::handle>::type pybind11::detail::type_caster<T, typename std::enable_if<(std::is_arithmetic<_Tp>::value && (! pybind11::detail::negation<std::is_same<pybind11::detail::bools<pybind11::detail::negation<std::is_same<CharT, char> >::value, pybind11::detail::negation<std::is_same<CharT, char16_t> >::value, pybind11::detail::negation<std::is_same<CharT, char32_t> >::value, pybind11::detail::negation<std::is_same<CharT, wchar_t> >::value, true>, pybind11::detail::bools<true, pybind11::detail::negation<std::is_same<CharT, char> >::value, pybind11::detail::negation<std::is_same<CharT, char16_t> >::value, pybind11::detail::negation<std::is_same<CharT, char32_t> >::value, pybind11::detail::negation<std::is_same<CharT, wchar_t> >::value> > >::value)), void>::type>::cast(U, pybind11::return_value_policy, pybind11::handle) [with U = U; T = double]’
  240 |     cast(U src, return_value_policy /* policy */, handle /* parent */) {
      |     ^~~~
/usr/local/include/pybind11/cast.h:240:5: note:   template argument deduction/substitution failed:
/usr/local/include/pybind11/cast.h: In substitution of ‘template<class U> static typename std::enable_if<(((! std::is_floating_point<_Tp>::value) && std::is_unsigned<T>::value) && (sizeof (U) > sizeof (long unsigned int))), pybind11::handle>::type pybind11::detail::type_caster<double, void>::cast<U>(U, pybind11::return_value_policy, pybind11::handle) [with U = const double (*)[3]]’:
/usr/local/include/pybind11/pybind11.h:296:40:   required from ‘void pybind11::cpp_function::initialize(Func&&, Return (*)(Args ...), const Extra& ...) [with Func = pybind11::class_<type_, options>::def_readwrite(const char*, D C::*, const Extra& ...) [with C = Ctest; D = double [3][3]; Extra = {}; type_ = Ctest; options = {}]::<lambda(const type&)>; Return = const double (&)[3][3]; Args = {const Ctest&}; Extra = {pybind11::is_method}]’
/usr/local/include/pybind11/pybind11.h:145:9:   required from ‘pybind11::cpp_function::cpp_function(Func&&, const Extra& ...) [with Func = pybind11::class_<type_, options>::def_readwrite(const char*, D C::*, const Extra& ...) [with C = Ctest; D = double [3][3]; Extra = {}; type_ = Ctest; options = {}]::<lambda(const type&)>; Extra = {pybind11::is_method}; <template-parameter-1-3> = void]’
/usr/local/include/pybind11/pybind11.h:1716:22:   required from ‘pybind11::class_<type_, options>& pybind11::class_<type_, options>::def_readwrite(const char*, D C::*, const Extra& ...) [with C = Ctest; D = double [3][3]; Extra = {}; type_ = Ctest; options = {}]’
1.cpp:150:54:   required from here
/usr/local/include/pybind11/cast.h:240:5: error: no type named ‘type’ in ‘struct std::enable_if<false, pybind11::handle>’
/usr/local/include/pybind11/pybind11.h: In instantiation of ‘void pybind11::cpp_function::initialize(Func&&, Return (*)(Args ...), const Extra& ...) [with Func = pybind11::class_<type_, options>::def_readwrite(const char*, D C::*, const Extra& ...) [with C = Ctest; D = double [3][3]; Extra = {}; type_ = Ctest; options = {}]::<lambda(const type&)>; Return = const double (&)[3][3]; Args = {const Ctest&}; Extra = {pybind11::is_method}]’:
/usr/local/include/pybind11/pybind11.h:145:9:   required from ‘pybind11::cpp_function::cpp_function(Func&&, const Extra& ...) [with Func = pybind11::class_<type_, options>::def_readwrite(const char*, D C::*, const Extra& ...) [with C = Ctest; D = double [3][3]; Extra = {}; type_ = Ctest; options = {}]::<lambda(const type&)>; Extra = {pybind11::is_method}; <template-parameter-1-3> = void]’
/usr/local/include/pybind11/pybind11.h:1716:22:   required from ‘pybind11::class_<type_, options>& pybind11::class_<type_, options>::def_readwrite(const char*, D C::*, const Extra& ...) [with C = Ctest; D = double [3][3]; Extra = {}; type_ = Ctest; options = {}]’
1.cpp:150:54:   required from here
/usr/local/include/pybind11/cast.h:244:5: note: candidate: ‘template<class T_, typename std::enable_if<std::is_same<double, typename std::remove_cv< <template-parameter-1-1> >::type>::value, int>::type <anonymous> > static pybind11::handle pybind11::detail::type_caster<T, typename std::enable_if<(std::is_arithmetic<_Tp>::value && (! pybind11::detail::negation<std::is_same<pybind11::detail::bools<pybind11::detail::negation<std::is_same<CharT, char> >::value, pybind11::detail::negation<std::is_same<CharT, char16_t> >::value, pybind11::detail::negation<std::is_same<CharT, char32_t> >::value, pybind11::detail::negation<std::is_same<CharT, wchar_t> >::value, true>, pybind11::detail::bools<true, pybind11::detail::negation<std::is_same<CharT, char> >::value, pybind11::detail::negation<std::is_same<CharT, char16_t> >::value, pybind11::detail::negation<std::is_same<CharT, char32_t> >::value, pybind11::detail::negation<std::is_same<CharT, wchar_t> >::value> > >::value)), void>::type>::cast(T_*, pybind11::return_value_policy, pybind11::handle) [with T_ = T_; typename std::enable_if<std::is_same<T, typename std::remove_cv<_Up>::type>::value, int>::type <anonymous> = <enumerator>; T = double]’
  244 |     PYBIND11_TYPE_CASTER(T, const_name<std::is_integral<T>::value>("int", "float"));
      |     ^~~~~~~~~~~~~~~~~~~~
/usr/local/include/pybind11/cast.h:244:5: note:   template argument deduction/substitution failed:
/usr/local/include/pybind11/cast.h:244:5: error: no type named ‘type’ in ‘struct std::enable_if<false, int>’
  244 |     PYBIND11_TYPE_CASTER(T, const_name<std::is_integral<T>::value>("int", "float"));
      |     ^~~~~~~~~~~~~~~~~~~~
In file included from /usr/local/include/pybind11/embed.h:12,
                 from 1.cpp:1:
/usr/local/include/pybind11/pybind11.h:144:5: error: ‘pybind11::cpp_function::cpp_function(Func&&, const Extra& ...) [with Func = pybind11::class_<type_, options>::def_readwrite(const char*, D C::*, const Extra& ...) [with C = Ctest; D = double [3][3]; Extra = {}; type_ = Ctest; options = {}]::<lambda(pybind11::class_<Ctest>::type&, const double (&)[3][3])>; Extra = {pybind11::is_method}; <template-parameter-1-3> = void]’, declared using local type ‘pybind11::class_<type_, options>::def_readwrite(const char*, D C::*, const Extra& ...) [with C = Ctest; D = double [3][3]; Extra = {}; type_ = Ctest; options = {}]::<lambda(pybind11::class_<Ctest>::type&, const double (&)[3][3])>’, is used but never defined [-fpermissive]
  144 |     cpp_function(Func &&f, const Extra &...extra) {
      |     ^~~~~~~~~~~~

**Comment by huqin-RM** [comment]
class Ctest
{
public:
    int m_vdMat[3][3] = {{1,2,3},{4,2,3},{1,8,3}};
};


PYBIND11_EMBEDDED_MODULE(fast_calc,m){


    py::class_<Ctest>(m, "Ctest")
            .def(py::init<>())
            .def("m_vdMat", [](Ctest &s){
                                             py::object sb = py::cast(s);
                                             pybind11::buffer_info info = pybind11::buffer_info(s.m_vdMat, sizeof(int), pybind11::format_descriptor<int>::format(), 2, {3,3}, {sizeof(int)*3, sizeof(int)});
                                             pybind11::array arr(info, sb.ptr());
                                             return  arr;
                                           });

**Comment by evguran** [comment]
Hi, attributes indeed defined via `def_readwrite`, but why do you refer to `CIntrinsicsM` class? 
`.def_readwrite("m_vdMat", &CIntrinsicsM::m_vdMat);`
I suppose you should also `#include "pybind11/stl.h"`
